File: scripts/CountRenamedFunctions.java
// Counts function naming progress in the current program.
// @category Imperialism/Stats

import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.Function;
import ghidra.program.model.symbol.SourceType;
import ghidra.program.model.symbol.Symbol;
import ghidra.util.exception.CancelledException;

public class CountRenamedFunctions extends GhidraScript {
	@Override
	protected void run() throws Exception {
		int total = 0;
		int userDefined = 0;
		int nonGeneric = 0;

		for (Function fn : currentProgram.getFunctionManager().getFunctions(true)) {
			if (monitor.isCancelled()) {
				throw new CancelledException();
			}

			total++;
			Symbol sym = fn.getSymbol();
			if (sym != null && sym.getSource() == SourceType.USER_DEFINED) {
				userDefined++;
			}

			String name = fn.getName();
			if (!name.startsWith("FUN_") && !name.startsWith("thunk_FUN_")) {
				nonGeneric++;
			}
		}

		println("PROGRAM=" + currentProgram.getName());
		println("TOTAL_FUNCTIONS=" + total);
		println("USER_DEFINED_FUNCTION_SYMBOLS=" + userDefined);
		println("NON_GENERIC_NAMES=" + nonGeneric);
	}
}

File: scripts/count_renamed_functions.py
#@category Imperialism/Stats

from ghidra.program.model.symbol import SourceType

fm = currentProgram.getFunctionManager()

total = 0
user_defined = 0
non_generic = 0

it = fm.getFunctions(True)
while it.hasNext():
    fn = it.next()
    total += 1

    sym = fn.getSymbol()
    if sym is not None and sym.getSource() == SourceType.USER_DEFINED:
        user_defined += 1

    name = fn.getName()
    if not name.startswith("FUN_") and not name.startswith("thunk_FUN_"):
        non_generic += 1

print("PROGRAM=%s" % currentProgram.getName())
print("TOTAL_FUNCTIONS=%d" % total)
print("USER_DEFINED_FUNCTION_SYMBOLS=%d" % user_defined)
print("NON_GENERIC_NAMES=%d" % non_generic)
